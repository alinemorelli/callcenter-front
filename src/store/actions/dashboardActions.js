import * as actionTypes from './actionsTypes'
import axios from '../../client/axios'

export const getDashboardDataSuccess = (activeCalls) => {
  return {
    type: actionTypes.GET_DASHBOARD_DATA,
    activeCalls: activeCalls
  }
}

export const getDashboardData = (url) => {
  return (dispatch) => {
    axios.get(url)
    .then(response => {
      dispatch(getDashboardDataSuccess(response.data))
    })
    .catch(error => {
      //TODO: handle error when implemented
    })
  }
}
