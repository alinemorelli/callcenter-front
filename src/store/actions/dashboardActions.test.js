import * as actions from './dashboardActions'
import * as types from './actionsTypes'
import { random, phone } from 'faker'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import fetchMock from 'fetch-mock'
import expect from 'expect'

const activeCalls = [{
  call_id: random.uuid(),
  their_number: phone.phoneNumber()
}]

describe('actions', () => {
  it('should create an action to get dashbaord data', () => {
    const expectedAction = {
      type: types.GET_DASHBOARD_DATA,
      activeCalls
    }
    expect(actions.getDashboardDataSuccess(activeCalls)).toEqual(expectedAction)
  })
})

describe('get dashboard data', () => {
  const middlewares = [thunk]
  const mockStore = configureMockStore(middlewares)
  const store = mockStore({})
  afterEach(() => {
    fetchMock.restore()
  })

  it('creates GET_DASHBOARD_DATA when fetching todos has been done', () => {
    fetchMock.getOnce('/dashboard', {
      body: { data: activeCalls },
      headers: { 'content-type': 'application/json' }
    })

    store.dispatch(actions.getDashboardData(activeCalls))
    expect(store.getActions()).toMatchSnapshot()
  })
})
