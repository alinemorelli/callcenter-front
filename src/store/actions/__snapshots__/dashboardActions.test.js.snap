// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get dashboard data creates GET_DASHBOARD_DATA when fetching todos has been done 1`] = `Array []`;
